service/service.go

```diff
@@ -2,162 +2,203 @@ package service
 
 import (
 	"context"
-	"time"
+	"fmt"
+	"net/http"
 
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/dp-import-cantabular-dimension-options/config"
 	"github.com/ONSdigital/dp-import-cantabular-dimension-options/event"
+	dpkafka "github.com/ONSdigital/dp-kafka/v2"
 	kafka "github.com/ONSdigital/dp-kafka/v2"
-	"github.com/ONSdigital/log.go/log"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/log.go/v2/log"
 	"github.com/gorilla/mux"
 	"github.com/pkg/errors"
 )
 
 // Service contains all the configs, server and clients to run the event handler service
 type Service struct {
-	server          HTTPServer
-	router          *mux.Router
-	serviceList     *ExternalServiceList
-	healthCheck     HealthChecker
-	consumer        kafka.IConsumerGroup
-	shutdownTimeout time.Duration
+	cfg         *config.Config
+	server      HTTPServer
+	healthCheck HealthChecker
+	consumer    kafka.IConsumerGroup
 }
 
-// Run the service
-func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCommit, version string, svcErrors chan error) (*Service, error) {
-	log.Event(ctx, "running service", log.INFO)
-
-	// Read config
-	cfg, err := config.Get()
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to retrieve service configuration")
+// GetKafkaConsumer returns a Kafka consumer with the provided config
+var GetKafkaConsumer = func(ctx context.Context, cfg *config.Config) (dpkafka.IConsumerGroup, error) {
+	cgChannels := dpkafka.CreateConsumerGroupChannels(1)
+	kafkaOffset := dpkafka.OffsetNewest
+	if cfg.KafkaOffsetOldest {
+		kafkaOffset = dpkafka.OffsetOldest
 	}
-	log.Event(ctx, "got service configuration", log.Data{"config": cfg}, log.INFO)
+	return dpkafka.NewConsumerGroup(
+		ctx,
+		cfg.KafkaAddr,
+		cfg.HelloCalledTopic,
+		cfg.HelloCalledGroup,
+		cgChannels,
+		&dpkafka.ConsumerGroupConfig{
+			KafkaVersion: &cfg.KafkaVersion,
+			Offset:       &kafkaOffset,
+		},
+	)
+}
 
-	// Get HTTP Server with collectionID checkHeader middleware
-	r := mux.NewRouter()
-	s := serviceList.GetHTTPServer(cfg.BindAddr, r)
+// GetHTTPServer returns an http server
+var GetHTTPServer = func(bindAddr string, router http.Handler) HTTPServer {
+	s := dphttp.NewServer(bindAddr, router)
+	s.HandleOSSignals = false
+	return s
+}
 
-	// Get Kafka consumer
-	consumer, err := serviceList.GetKafkaConsumer(ctx, cfg)
+// GetHealthCheck returns a healthcheck
+var GetHealthCheck = func(cfg *config.Config, buildTime, gitCommit, version string) (HealthChecker, error) {
+	versionInfo, err := healthcheck.NewVersionInfo(buildTime, gitCommit, version)
 	if err != nil {
-		log.Event(ctx, "failed to initialise kafka consumer", log.FATAL, log.Error(err))
 		return nil, err
```

Comment:
Again, probably good to add context to the error here. Will add to cantabular-dataset service also

Reply:
also the message is wrong!

Reply:
I think this one is sorted (this is the old code) - for the new code we simple return the error